Reader: Add hover transition Comment and Edit buttons

jancavan opened the pull request:
For Reader action buttons, the Edit and Comments button don't have hover transitions while the others do. This PR fixes that.

![screenshot 2016-09-23 15 04 55](https://cloud.githubusercontent.com/assets/4924246/18802669/1ab20930-819f-11e6-9175-5780ba5d42f7.png)

bluefuton commented on client/blocks/post-edit-button/index.jsx:
```diff
@@ -13,10 +13,10 @@ import { getEditURL } from 'lib/posts/utils';
 	const editUrl = getEditURL( post, site );
 	return (
-		<a className="post-edit-button" href={ editUrl } onClick={ onClick }>
+		<span className="post-edit-button" href={ editUrl } onClick={ onClick }>
```
Not sure about the change to `span` here.

It's a straightforward link so `a` is the right choice semantically I think.

fraying commented:
Unless I'm hallucinating, which is a possibility, the hover transitions on fullpost fade in, whereas, on the main stream, they change immediately. They should match. Could we change the fullpost hover to match the one on the main stream?

jancavan commented:
> the hover transitions on fullpost fade in, whereas, on the main stream, they change immediately.

Hmm, that's odd. I see transitions on both main stream and full-post. Is it a particular button or all of them?

fraying commented:
Huh. Weird. We're talking about the card footer links on comments and likes, right? On live wordpress.com, I see them change instantly on hover.

Personally, I like the instant change. The fade-in hover makes me feel like my computer is slow.

jancavan commented:
@fraying Actually, sorry my PR subject is incorrect, just updated it.

Yes, the action links (Edit, Share, Comments, Like, Ellipsis). The transitions have always been there except for Comments (stream) and Comments/Edit (full-post). I think it was just an oversight so I made them all match.

> Personally, I like the instant change.

I don't really have a preference, but since we don't use any transitions on any links, they can be removed for uniformity.

bluefuton commented:
Looks good to me 🚢

jancavan commented:
I have removed hover transitions in: https://github.com/Automattic/wp-calypso/pull/8183/commits/42b54ac7c7ddcca4005d6d2da1b0896d12cd28aa

Rebased this branch, but not sure what happened that it misaligned the Share button. Fixed in: https://github.com/Automattic/wp-calypso/pull/8183/commits/b9337cf45b67b00ca3fe4072ad8a9735823781f0
